### PR TITLE
Performance Enhancements

### PR DESCRIPTION
**Pull request by therobbiedavis**
# Changed
- Changed: (Performance) Added a short circuit in CanOpen to check for rar files and force SharpCompress for deflating. This avoids an IO and speeds up processing.
- Changed: (Performance) Added more Split Queries for many queries that are JOINs, speeding them up significantly

### CODE REVIEW

File: API/Services/ArchiveService.cs [hunk 1]
```diff
@@ -57,12 +57,8 @@ public virtual ArchiveLibrary CanOpen(string archivePath)
...
             if (string.IsNullOrEmpty(archivePath) || !(File.Exists(archivePath) && Parser.Parser.IsArchive(archivePath) || Parser.Parser.IsEpub(archivePath))) return ArchiveLibrary.NotSupported;
 
-            try
-            {
-                using var a2 = ZipFile.OpenRead(archivePath);
-                return ArchiveLibrary.Default;
-            }
-            catch (Exception)
+            var ext = _directoryService.FileSystem.Path.GetExtension(archivePath);
+            if (ext == ".cbr" || ext == ".rar")
```
Comment (majora2007) [inline API/Services/ArchiveService.cs:61]: Change this to if (ext == ".cbr" || ext == ".rar") return ArchiveLibrary.SharpCompress

Comment (majora2007) [reply API/Services/ArchiveService.cs:61]: Also make sure you do a .toLower() (actually .toUpper() would be better) because some files are .RAR

Comment (therobbiedavis) [reply API/Services/ArchiveService.cs]: Resolve in commit 731d078

File: API/Services/ArchiveService.cs [hunk 2]
```diff
@@ -57,6 +57,9 @@ public virtual ArchiveLibrary CanOpen(string archivePath)
         {
             if (string.IsNullOrEmpty(archivePath) || !(File.Exists(archivePath) && Parser.Parser.IsArchive(archivePath) || Parser.Parser.IsEpub(archivePath))) return ArchiveLibrary.NotSupported;
 
+            var ext = _directoryService.FileSystem.Path.GetExtension(archivePath);
+            if (ext.ToUpper() == ".CBR" || ext.ToUpper() == ".RAR") return ArchiveLibrary.SharpCompress;
```
Comment (majora2007) [inline API/Services/ArchiveService.cs:61]: This just needs to be .Equals()

**Review by majora2007 — CHANGES_REQUESTED** [review]
Just one comment, then it's good

File: API/Services/ArchiveService.cs (hunk 2 above)
Comment (therobbiedavis) [reply API/Services/ArchiveService.cs]: Resolved with 81c0834

**Pull request merged**